vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors_moe.py

```diff
@@ -269,9 +271,14 @@ def create_weights(self, layer: torch.nn.Module, num_experts: int,
                        hidden_size: int, intermediate_size: int,
                        params_dtype: torch.dtype, **extra_weight_attrs):
 
+        assert params_dtype == torch.float16, (
+            "float16 is required for MoE compressd models. Set dtype=torch.float16"  # noqa: E501
+        )
+
         # Will transpose the loaded weight along the
         # intermediate and hidden dim sizes. Will
         # shard for TP along the transposed dims
+        intermediate_full = extra_weight_attrs.pop("intermediate_full")
```

Comment:
nit: maybe rename `intermediate_size` -> `intermediate_size_per_partition` and `intermediate_full` -> `intermediate_size`?
This would make the names consistent with other quant configs, e.g. `vllm/model_executor/layers/quantization/gptq.py`